blockchain/validate.go

```diff
@@ -717,6 +864,22 @@ func checkBlockSanity(block *dcrutil.Block, timeSource MedianTimeSource, flags B
 		case stake.TxTypeSSRtx:
 			totalRevocations++
 		}
+
+		// If treasury agenda is enabled do the same type of math as
+		// above.
+		if isTreasuryEnabled {
+			switch txType {
+
+			case stake.TxTypeTAdd:
+				totalTreasury++
+
+			case stake.TxTypeTSpend:
+				totalTreasury++
```

Comment:
Realized now that while other consensus rules prevent it from happening, this specific check could allow TSPends to be mined before SVH. Maybe for the purposes of the SVH check below, TSPends should be tracked separately here.